cmd/filestransfer/filestransfer.go

```diff
@@ -0,0 +1,164 @@
+// Example: No connection pool
+//
+// This example shows how to transfer files to staging area, from which data can be loaded into snowflake
+// database tables. Apart from sending files to staging area using PUT command, files can also be downloaded
+// using GET command.
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"database/sql"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	sf "github.com/snowflakedb/gosnowflake"
+)
+
+func getDSN() (string, *sf.Config, error) {
+	env := func(key string, failOnMissing bool) string {
+		if value := os.Getenv(key); value != "" {
+			return value
+		}
+		if failOnMissing == true {
+			log.Fatalf("#{key} environment variable not set")
+		}
+		return ""
+	}
+	account := env("SNOWFLAKE_TEST_ACCOUNT", true)
+	user := env("SNOWFLAKE_TEST_USER", true)
+	password := env("SNOWFLAKE_TEST_PASSWORD", true)
+	database := env("SNOWFLAKE_TEST_DATABASE", true)
+	warehouse := env("SNOWFLAKE_TEST_WAREHOUSE", true)
+	host := env("SNOWFLAKE_TEST_HOST", true)
+	schema := env("SNOWFLAKE_TEST_SCHEMA", true)
+	portStr := env("SNOWFLAKE_TEST_PORT", false)
+	protocol := env("SNOWFLAKE_TEST_PROTOCOL", false)
+
+	port := 443
+	var err error
+	if len(portStr) > 0 {
+		port, err = strconv.Atoi(portStr)
+		if err != nil {
+			return "", nil, err
+		}
+	}
+
+	cfg := &sf.Config{
+		Account:   account,
+		User:      user,
+		Password:  password,
+		Database:  database,
+		Warehouse: warehouse,
+		Schema:    schema,
+		Host:      host,
+		Port:      port,
+		Protocol:  protocol,
+	}
+
+	dsn, err := sf.DSN(cfg)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return dsn, cfg, nil
+}
+
+func decompressAndRead(file *os.File) (string, error) {
+	gzipReader, err := gzip.NewReader(file)
+	defer gzipReader.Close()
+	if err != nil {
+		return "", err
+	}
+	var b bytes.Buffer
+	_, err = b.ReadFrom(gzipReader)
+	if err != nil {
+		return "", err
+	}
+	return b.String(), nil
+}
+
+func printRows(rows *sql.Rows) {
+	i := 1
+	for rows.Next() {
+		var col1, col2 string
+		if err := rows.Scan(&col1, &col2); err != nil {
+			log.Fatalf("error while scaning rows; err: %v", err)
+		}
+		fmt.Printf("Row %v: %v, %v\n", i, col1, col2)
+		i++
+	}
+}
+
+func main() {
+	if !flag.Parsed() {
+		flag.Parse()
+	}
+
+	//Opening connection
+	dsn, cfg, err := getDSN()
+	if err != nil {
+		log.Fatalf("error while creating DSN from config: %v, error: %v", cfg, err)
+	}
+	db, err := sql.Open("snowflake", dsn)
+
+	//Creating table to which the data from CSV file will be copied
+	_, err = db.Exec("CREATE OR REPLACE TABLE TEST_FILE_DATA(num integer, text varchar);")
+	if err != nil {
+		log.Fatalf("error while creating table; err: %v", err)
+	}
+	defer db.Close()
+
+	//Uploading data_to_upload.csv to internal stage for table TEST_FILE_DATA
+	currentDir, _ := os.Getwd()
```

Comment:
Does this function returns the path to the app file directory or caller? Based on the next line I presume the second case and it would be very fragile (works only when invoking from repo root?). Also, what happens when we install app in GOPATH? I don't see copy this file in makefile.

Reply:
Nice catch. Changed it to storing file content in code and then saving it to tmp file using os.CreateTemp(...). I was considering passing path as an argument but decided these examples should be as easy to run as possible.